iocage/lib/Jail.py

```diff
@@ -0,0 +1,500 @@
+import JailConfig
+import Network
+import Storage
+import Releases
+import Release
+import helpers
+
+import ZFSBasejailStorage
+import ZFSShareStorage
+import NullFSBasejailStorage
+import StandaloneJailStorage
+
+import subprocess
+import uuid
+
+
+class Jail:
+
+    def __init__(self, data={}, zfs=None, host=None, logger=None, new=False):
+
+        helpers.init_logger(self, logger)
+        helpers.init_zfs(self, zfs)
+        helpers.init_host(self, host)
+
+        if isinstance(data, str):
+            data = {"uuid": self._resolve_uuid(data)}
+
+        self.config = JailConfig.JailConfig(
+            data=data, jail=self, logger=self.logger)
+
+        self.networks = []
+
+        self.storage = Storage.Storage(
+            auto_create=True, safe_mode=False,
+            jail=self, logger=self.logger, zfs=self.zfs)
+
+        self.jail_state = None
+
+        if new is False:
+            self.config.read()
+        # self.update_jail_state()
+
+    @property
+    def zfs_pool_name(self):
+        return self.host.datasets.root.name.split("/", maxsplit=1)[0]
+
+    def start(self):
+        self.require_jail_existing()
+        self.require_jail_stopped()
+
+        release = self.release
+
+        NullFSBasejailStorage.NullFSBasejailStorage.umount_nullfs(
+            self.storage)
+
+        storage_backend = None
+
+        if self.config.basejail_type == "zfs":
+            storage_backend = ZFSBasejailStorage.ZFSBasejailStorage
+
+        if self.config.basejail_type == "nullfs":
+            storage_backend = NullFSBasejailStorage.NullFSBasejailStorage
+
+        # if self.config.type == "clonejail":
+        #   pass
+
+        if storage_backend is not None:
```

Comment:
flake8 was complaining about it. we can change the setting, so that I can omit the explicit check for None.